LDAP-Auth/LdapProfileImageSyncTask.cs

```diff
@@ -97,7 +100,24 @@ public async Task ExecuteAsync(IProgress<double> progress, CancellationToken can
                     continue;
                 }
 
-                var ldapProfileImage = ldapAuthProvider.GetAttribute(ldapUser, ProfileImageAttr)?.ByteValue;
+                byte[] ldapProfileImage = null;
+                if (ProfileImageAttrFormat == "base64")
+                {
+                    ldapProfileImage = Convert.FromBase64String(ldapAuthProvider.GetAttribute(ldapUser, ProfileImageAttr)?.StringValue);
+                }
+                else if (ProfileImageAttrFormat == "binary")
+                {
+                    ldapProfileImage = ldapAuthProvider.GetAttribute(ldapUser, ProfileImageAttr)?.ByteValue;
+                }
+                else if (ProfileImageAttrFormat == "url")
+                {
+                    using var client = new HttpClient();
+                    ldapProfileImage = await client.GetByteArrayAsync(ldapAuthProvider.GetAttribute(ldapUser, ProfileImageAttr)?.StringValue);
+                }
+                else
+                {
+                    _logger.LogError("Unknown profile image format: {Format}", ProfileImageAttrFormat);
+                }
```

Comment:
It would probably be cleaner if this could be done via a match pattern. If the match returns a `Task<byte[]>` it should be simple to return `Task.FromResult` in the first two cases and just use a async method that retrieves the picture from the net in the third.

Additionally I highly recommend against using a stringly typed Format and instead use an Enum.